### ISSUE TITLE
`W1116 (isinstance-second-argument-not-valid-type)`  false positive for union types

### ISSUE THREAD

**Issue by randolf-scholz** [PyCQA/pylint#8205]
### Bug description

```python
assert isinstance(0, int | str)  # ✘ raises W1116
```

### Command used

```shell
pylint
```

### Pylint output

```shell
************* Module isinstance_second_argument_not_valid_type
bugs/pylint/isinstance_second_argument_not_valid_type.py:1:0: C0114: Missing module docstring (missing-module-docstring)
bugs/pylint/isinstance_second_argument_not_valid_type.py:5:7: W1116: Second argument of isinstance is not a type (isinstance-second-argument-not-valid-type)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

Union types are supported by `isinstance` and `issubclass` since 3.10 https://docs.python.org/3/library/functions.html#isinstance

### Pylint version

```shell
pylint 2.16.1
astroid 2.14.1
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```
